src/ui/components/Overlay/index.js

```diff
@@ -39,8 +49,12 @@ export default class Overlay extends React.Component {
       <div className={classNames('Overlay', className, {
         'Overlay--visible': this.state.visible,
       })} ref={(ref) => { this.overlayContainer = ref; }}>
-        <div className="Overlay-background"
-          ref={(ref) => { this.overlayBackground = ref; }} />
+        <div
+          onClick={this.onClickBackground}
+          onTouchEnd={this.onClickBackground}
```

Comment:
A tap will fire the touch events but also the mouse events including click (unless you `preventDefault` somewhere higher in the chain, with click being the last event)